syntax = "proto3";

package aoeinterpolator;

service AoeInterpolator
{
    rpc StartInterpolator(StartInterpolatorRequest) returns (StartInterpolatorResponse) { }
    rpc GetInterpolatedFrame(InterpolatedFrameRequest) returns (InterpolatedFrameResponse) { }
    rpc StopInterpolator(StopInterpolatorRequest) returns (StopInterpolatorResponse) { }
    rpc TerminateScript(TerminateScriptRequest) returns (TerminateScriptResponse) { }
}

message StartInterpolatorRequest
{
    bool allowCuda = 1;
    int32 numThreads = 2;
}

message StartInterpolatorResponse
{
    bool success = 1;
    string message = 2;
}

message InterpolatedFrameRequest
{
    bytes frame1 = 1;
    bytes frame2 = 2;
    int32 transparentR = 3;
    int32 transparentG = 4;
    int32 transparentB = 5;
    float alpha = 6;
}

message InterpolatedFrameResponse
{
    bytes frame = 1;
}

message StopInterpolatorRequest
{

}

message StopInterpolatorResponse
{
    bool success = 1;
    string message = 2;
}

message TerminateScriptRequest
{

}

message TerminateScriptResponse
{
    bool success = 1;
    string message = 2;
}
